event/event_test.go

```diff
@@ -60,141 +101,242 @@ var newTests = []struct {
 		eventpkg.SystemEventReceivedData{},
 		eventpkg.Event{
 			EventType:          eventpkg.Type("user.created"),
-			CloudEventsVersion: eventpkg.TransformationVersion,
+			CloudEventsVersion: "0.1",
 			Source:             "https://serverless.com/event-gateway/#transformationVersion=0.1",
 			ContentType:        "application/json",
 			Data:               eventpkg.SystemEventReceivedData{},
 			Extensions: zap.MapStringInterface{
 				"eventgateway": map[string]interface{}{
-					"transformed": true,
+					"transformed":            true,
 					"transformation-version": eventpkg.TransformationVersion,
 				},
 			},
 		},
 	},
+}
+
+var encodingTests = []struct {
+	body         []byte
+	contentType  string
+	expectedBody interface{}
+}{
+	{
+		[]byte("some=thing"),
+		"application/octet-stream",
+		[]byte("some=thing"),
+	},
 	{
-		// valid CloudEvent
-		eventpkg.Type("user.created"),
+		[]byte("some=thing"),
+		"application/x-www-form-urlencoded",
+		"some=thing",
+	},
+	{
+		[]byte("--X-INSOMNIA-BOUNDARY\r\nContent-Disposition: form-data; name=\"some\"\r\n\r\nthing\r\n--X-INSOMNIA-BOUNDARY--\r\n"),
+		"multipart/form-data; boundary=X-INSOMNIA-BOUNDARY",
+		"--X-INSOMNIA-BOUNDARY\r\nContent-Disposition: form-data; name=\"some\"\r\n\r\nthing\r\n--X-INSOMNIA-BOUNDARY--\r\n",
+	},
+	{
+		[]byte(`{"hello": "world"}`),
 		"application/json",
-		[]byte(`{
+		map[string]interface{}{"hello": "world"},
+	},
+}
+
+var fromRequestTests = []struct {
+	url           string
```

Comment:
url doesn't matter here, we can remove it.

Reply:
Well, it actually does. Because `NewHTTPRequestData` expects a URL.

Reply:
In the `http.Request` that we're passing to `FromRequest` function